src/inference_profiling/profile_program_local.py

```diff
@@ -0,0 +1,89 @@
+# This script is used to profile the performance of test_models.py file to determine the performance
+# parameters for predictions made on pytorch models on a local environment. For more details check
+# check out the script "test_models.py "
+
+# Import the required utilities
+import psutil as ps
+import time
+from subprocess import PIPE
+import os
+import csv
+from argparse import ArgumentParser
+
+# Create a parser
+parser = ArgumentParser()
+
+parser.add_argument('--batch', required=True)
+parser.add_argument('--model_name', required=True)
+
+args = parser.parse_args()
+
+# define the command for the subprocess
+model_name = args.model_name
+batch = args.batch
+
+cmd = ["python", "test_models.py"]+[batch, model_name]
+print(cmd)
+
+# Start the timer
+start_time = time.time()
+
+# Create the process
+process = ps.Popen(cmd, stdout=PIPE)
+
+print("The process ID is", process)
+
+# Initialize the variables.
+peak_sys_mem = 0
+peak_sys_cpu = 0
+peak_proc_mem = 0
+peak_proc_cpu = 0
+
+# while the process is running calculate resource utilization.
+print("Process is in the running state.")
+
+while (process.is_running()):
+    # set the sleep time to monitor at an interval of every second.
+    time.sleep(0.5)
```

Comment:
Just wondering why is it 0.5 here and 1 above?

Reply:
It was generally set as the execution time of test_models.py is pretty less as compared to concurrent_inference.py due to which while profiling the program, we can try to get performance parameters more accurately.